src/Simulation/Core/IQuantumMachineOutput.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+
+namespace Microsoft.Quantum.Runtime
+{
+    /// <summary>
+    /// Interface to access the results of a program executed in a quantum machine.
+    /// </summary>
+    public interface IQuantumMachineOutput
+    {
+
+        /// <summary>
+        /// Gets an output probabilistically.
+        /// Calls to this method are not deterministic, each call can return a different output.
+        /// </summary>
+        object GetOutput();
```

Comment:
It seems like this could be accomplished by randomly sampling `Histogram`. Does it need to be a separate method in the interface?

Reply:
It could be obtained by doing that, but it needs to be proportional to the frequency of the specific output. That requires a little bit of logic which is provided by this convenience so that the developer using the interface does not have to write that logic.